datahub-web-react/src/app/permissions/policy/policyUtils.ts

```diff
@@ -126,3 +127,106 @@ export const setFieldValues = (
     }
     return { ...filter, criteria: [...restCriteria, createCriterion(resourceFieldType, fieldValues)] };
 };
+
+
+/**
+ * Add an entry to the ListPolicies cache.
+ */
+export const addToListPoliciesCache = (client, newPolicy, pageSize, query) => {
+    // Read the data from our cache for this query.
+    const currData: ListPoliciesQuery = client.readQuery({
+        query: ListPoliciesDocument,
+        variables: {
+            input: {
+                start: 0,
+                count: pageSize,
+                query,
+            },
+        },
+    });
+
+    const completeNewPolicy = {
+        ...newPolicy,
+        actors:{
+            ...newPolicy?.actors,
+            roles : newPolicy?.actors?.roles || [],
+            resourceOwnersTypes: newPolicy?.actors?.resourceOwnersTypes || [],
+            resolvedOwnershipTypes: newPolicy?.actors?.resolvedOwnershipTypes || [],
+            resolvedUsers: newPolicy?.actors?.resolvedUsers || [],
+            resolvedGroups: newPolicy?.actors?.resolvedGroups || [],
+            resolvedRoles: newPolicy?.actors?.resolvedRoles || [],
+        },
+        resources : {
+            ...newPolicy.resources,
+            type: newPolicy?.resources?.type || '', // replace '' with the default value for type
```

Comment:
why are we overwriting the newPolicy stuff?

shouldn't the defaults come BEFORE the new policy is applied?

Reply:
giving error "missing type , resources"  in client.writeQuery